### PR TITLE
Fix variants filtering using Decipher criterion

### LINKED ISSUE #2993: pymongo.errors.OperationFailure: $and/$or/$nor must be a nonempty array
2021-11-24 16:40:55,296 - scout.server.app - ERROR: Exception on /cust003/85-WGS/sv/variants [POST] [in /home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py:1892]
Traceback (most recent call last):
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/hiseq.clinical/SERVER/apps/scout/git/scout/scout/server/utils.py", line 39, in decorated_function
    context = func(*args, **kwargs)
  File "/home/hiseq.clinical/SERVER/apps/scout/git/scout/scout/server/blueprints/variants/views.py", line 271, in sv_variants
    result_size = store.count_variants(case_obj["_id"], form.data, None, category)
  File "/home/hiseq.clinical/SERVER/apps/scout/git/scout/scout/adapter/mongo/variant.py", line 226, in count_variants
    return self.variant_collection.count_documents(query)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/collection.py", line 1786, in count_documents
    _cmd, self._read_preference_for(session), session)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/mongo_client.py", line 1471, in _retryable_read
    return func(session, server, sock_info, slave_ok)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/collection.py", line 1780, in _cmd
    sock_info, slave_ok, cmd, collation, session)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/collection.py", line 1677, in _aggregate_one_result
    session=session)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/collection.py", line 253, in _command
    user_fields=user_fields)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/pool.py", line 694, in command
    exhaust_allowed=exhaust_allowed)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/network.py", line 161, in command
    parse_write_concern_error=parse_write_concern_error)
  File "/home/hiseq.clinical/miniconda/envs/prod/lib/python3.6/site-packages/pymongo/helpers.py", line 164, in _check_command_response
    raise OperationFailure(errmsg, code, response, max_wire_version)
pymongo.errors.OperationFailure: $and/$or/$nor must be a nonempty array, full error: {'operationTime': Timestamp(1637768444, 1), 'ok': 0.0, 'errmsg': '$and/$or/$nor must be a nonempty array', 'code': 2, 'codeName': 'BadValue', '$clusterTime': {'clusterTime': Timestamp(1637768444, 1), 'signature': {'hash': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', 'keyId': 0}}}

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix variants filtering using the Decipher checkbox (fix #2993)

**How to test**:
1. Locally, on master branch, go to SVs variants and filter by clicking on the Decipher checkbox
2. The page should crash with the error described in #2993 
3. Repeat from this branch

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
This is ready for review. We should include it in the new release I think!

File: scout/adapter/mongo/query.py [hunk 1]
```diff
@@ -168,7 +168,6 @@ def build_query(
 
         These are the different query options:
             {
-                'genetic_models': list,
```
Comment (dnil) [inline scout/adapter/mongo/query.py:171]: 👍 👍

File: scout/adapter/mongo/query.py [hunk 2]
```diff
@@ -296,7 +295,7 @@ def build_query(
             secondary_filter = self.secondary_query(query, mongo_query)
             # If there are no primary criteria given, all secondary criteria are added as a
             # top level '$and' to the query.
-            if primary_terms is False:
+            if secondary_filter and primary_terms is False:
```
Comment (dnil) [inline scout/adapter/mongo/query.py:298]: 💯

Comment (dnil) [reply scout/adapter/mongo/query.py:298]: Should be fine, as there is already a check for existing `$and` in the coordinate query below.

**Review by dnil — APPROVED** [review]

**Pull request merged**